lib/realtime_web/controllers/tenant_controller.ex

```diff
@@ -232,6 +237,7 @@ defmodule RealtimeWeb.TenantController do
 
   def reload(conn, %{"tenant_id" => tenant_id}) do
     Logger.metadata(external_id: tenant_id, project: tenant_id)
+    Tracer.set_attributes(external_id: tenant_id)
```

Comment:
you can do this all in a plug in the router btw to cover request

Reply:
Sadly I don't think we can if we want to hook on the opentelemetry_phoenix span . opentelemetry_phoenix creates the span after the router dispatches so there is no span at the router stage to add an attribute to. 

At least I tried and it always showed that no span context exists. Unless we create a span before opentelemetry_phoenix does.


https://github.com/open-telemetry/opentelemetry-erlang-contrib/blob/70f561d4f92f9233ade373351d0e97287588eb30/instrumentation/opentelemetry_phoenix/lib/opentelemetry_phoenix.ex#L151-L161

Reply:
I will give it a try again though. Now that I have the integration tests to quickly test this out.

Reply:
huh I was probably holding it wrong... I was adding it on the "AssignTenant" plug and it was always showing no span context? 🤔 Well 🤷 Will update there and remove all the controller changes!

Thanks for bringing it up!

Reply:
Yeah but this specific one (AssignTenant) does it based on the host (https://$project_id.supabase.com) while that other one uses the path_params. So not a single solution across all endpoints. 🤷